### PR TITLE
Change oshinko-cli to reference local "core" directory for oshinko core

### PR DESCRIPTION
**Pull request by tmckayus**
This change is the first step in co-locating oshinko-cli, oshinko-core, and oshinko-rest so that the code is easier to keep in sync.  A copy of oshinko-core which works with the vendor dir from origin 1.4.1 has been put in the oshinko-cli repo in the "core" directory.  The oshinko-core dir in vendor has been removed, and the imports etc have been updated.

### CODE REVIEW

**Comment by sub-mod** [comment]
@tmckayus  thanks
imo it is better to make this PR a single commit.there are 11 commits for this PR.

**Comment by tmckayus** [comment]
That is a side effect of the process used to pull the files over and preserve the git history.  I don't know of a way to compress the commits and preserve the history ...

**Comment by sub-mod** [comment]
can we just manually copy paste the code. I am not sure if git history is important.

**Comment by tmckayus** [comment]
I think history is important.  We are going to do the same thing for oshinko-rest as well, where arguably the history is more extensive.  We should keep it.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (elmiko) [inline .gitignore]: looks like some merge kruft got left in the ignore file

**Comment by elmiko** [comment]
tested locally, works for me. thanks Trevor!

**Pull request merged**